### PR TITLE
[DNM]ADAMANT TAKE ONE

### PR DESCRIPTION
**Pull request by Thunder12345**
New boss and shit, only here to be testmerged before I go away for feedback purposes

### CODE REVIEW

**Comment by ChangelingRain** [comment]
ADAMANT's sprite is 64x64 and should be in an appropriately-sized DMI.

File: code/modules/mob/living/simple_animal/hostile/megafauna/adamant.dm [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+/mob/living/simple_animal/hostile/megafauna/adamant/proc/projectile_burst()
+	playsound(get_turf(src), 'sound/magic/clockwork/invoke_general.ogg', 200, 1, 2)
+	for(var/turf/T in range(2, target))
+		if(prob(30))
+			shoot_projectile(T)
+	sleep(20)
+
+
+/mob/living/simple_animal/hostile/megafauna/adamant/proc/telefrag(mob/living/target)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/megafauna/adamant.dm]: Having an argument with the same name as one of the mob's vars is a bad idea and might cause weird issues.

**Comment by Thunder12345** [comment]
The final version's going to be 96x96, so I'm putting it there to save having to change it later. Literally everything is just placeholder sprites while I make sure the code/gameplay are okay

**Comment by GunHog** [comment]
So it is NOT going to be the giant death mech shown on your avatar? I was hoping that if it was, we could get mining borg upgrades included in the loot, so they get something too!

**Comment by Thunder12345** [comment]
In the end no, that's just a thing I'm using as a placeholder

**Comment by Bawhoppen** [comment]
I still have more ideas for you but keyboard is broken right now :frowning:

File: code/modules/mob/living/simple_animal/hostile/megafauna/adamant.dm [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+	var/state = STATE_NORMAL
+	var/obj/item/device/gps/internal
+	var/struck = 0
+
+
+/obj/item/device/gps/internal/colossus
+	icon_state = null
+	gpstag = "Hardened Signal"
+	desc = "It's a signal, how can it be hard?"
```
Comment (Xhuis) [inline code/modules/mob/living/simple_animal/hostile/megafauna/adamant.dm]: ( ͡° ͜ʖ ͡°)

**Comment by optimumtact** [comment]
I don't see a need for more bosses than we already have.

**Comment by Kromgar** [comment]
WE NEED MORE EVERYTHING

File: code/modules/mob/living/simple_animal/hostile/megafauna/adamant.dm [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+			shoot_projectile(T)
+	sleep(20)
+
+
+/mob/living/simple_animal/hostile/megafauna/adamant/proc/telefrag(mob/living/victim)
+	var/matrix/ntransform = matrix(transform)
+	animate(src, transform=ntransform.Scale(10), alpha=0, 10)
+	density = 0
+	sleep(50)
+	freeze()
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/megafauna/adamant.dm]: The rest of this proc will wait for the freeze to finish, presumably you don't want that and it should be below the for loop.

**Comment by KorPhaeron** [comment]
>I don't see a need for more bosses than we already have.

This one doesn't even have original attacks, it's mostly copypasted code from other bosses except all the attacks kill you in one hit.

**Comment by KorPhaeron** [comment]
Anyway

-Copypasted Code
-Copypasted design
-No sprites

I'm not 100% against new bosses but they've got to do something new, or at least make an attempt at code standards

**Pull request closed without merging**